### PR TITLE
add vertex_ai/text-embedding-preview-0409

### PR DESCRIPTION
**Pull request by Dev-Khant**
Solves #2916.

### CODE REVIEW

**Comment by widike** [comment]
This would be very helpful to have.

**Comment by Dev-Khant** [comment]
@krrishdholakia @ishaan-jaff Can you please review this? Thanks.

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
Is this all that's required @Dev-Khant ? Can you send a screenshot of this working for you locally ?

File: litellm/model_prices_and_context_window_backup.json [hunk 1]
```diff
@@ -1144,6 +1144,26 @@
...
     },
+    "text-embedding-preview-0409": {
+        "max_tokens": 3072,
+        "max_input_tokens": 3072,
+        "output_vector_size": 768,
+        "input_cost_per_token": 0.00000000625,
+        "output_cost_per_token": 0,
+        "litellm_provider": "vertex_ai-embedding-models",
+        "mode": "embedding",
+        "source": "https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models#foundation_models"
```
Comment (ghost) [inline litellm/model_prices_and_context_window_backup.json:1155]: @Dev-Khant the source is incorrect here - i don't see `text-embedding-preview-0409` on https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models#foundation_models

Can you please link the source you got the pricing information from 


Also - is the output vector size only 768? On the example here I see 256 - https://cloud.google.com/blog/products/ai-machine-learning/google-cloud-announces-new-text-embedding-models

Comment (widike) [reply litellm/model_prices_and_context_window_backup.json:1155]: The model names for embeddings are correct. See here:
- https://cloud.google.com/blog/products/ai-machine-learning/google-cloud-announces-new-text-embedding-models
- https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings

Comment (ghost) [reply litellm/model_prices_and_context_window_backup.json:1155]: hey @demux79 thanks - do you have a link to the pricing for this too?

Comment (ghost) [reply litellm/model_prices_and_context_window_backup.json:1155]: is it this? https://cloud.google.com/vertex-ai/generative-ai/pricing

the cost of a multimodal embeddings api 
Screenshot 2024-04-15 at 4 53 53 PM

Comment (widike) [reply litellm/model_prices_and_context_window_backup.json:1155]: The first link also contains the pricing, right above "Dynamic embedding dimensions" (256 vs 768):

The pricing for our text embedding models is $0.000025/1,000 characters for [online requests](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/get-text-embeddings#latest_models) and $0.00002/1,000 characters for [batch requests](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/batch-prediction-genai-embeddings).

Comment (ghost) [reply litellm/model_prices_and_context_window_backup.json:1155]: ok - merging. will fix this on our end.

**Pull request merged**